File: TaylorCode/app/src/main/java/test/taylor/com/taylorcode/data_persistence/Activity.java
package test.taylor.com.taylorcode.data_persistence;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;
import androidx.annotation.NonNull;

@Entity(tableName = "taylor")
@TypeConverters(DataConverter.class)
public class Activity {
    @ColumnInfo(name = "user_name")
    private String userName;
    @ColumnInfo(name = "user_avatar_url")
    private String userAvatarUrl;
    private String title;
    @TypeConverters(DataConverter.class)
    private String time;
    private String type;
    @PrimaryKey
    @NonNull
    private String id;

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getUserAvatarUrl() {
        return userAvatarUrl;
    }

    public void setUserAvatarUrl(String userAvatrarUrl) {
        this.userAvatarUrl = userAvatrarUrl;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}

File: TaylorCode/app/src/main/java/test/taylor/com/taylorcode/data_persistence/Activity.kt
package test.taylor.com.taylorcode.data_persistence

import android.provider.Telephony
import androidx.room.TypeConverters
import test.taylor.com.taylorcode.data_persistence.DataConverter
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "taylor")
@TypeConverters(DataConverter::class)
class Activity {
    @ColumnInfo(name = "user_name")
    var userName: String? = null

    @ColumnInfo(name = "user_avatar_url")
    var userAvatarUrl: String? = null
    var title: String? = null

    @TypeConverters(DataConverter::class)
    var time: String? = null
    var type: String? = null

    @PrimaryKey
    var id: String = ""

    @TypeConverters(TestBeanConverter::class)
    var textBean: TestBean? = null

//    @TypeConverters(TestBean2Converter::class)
    @TypeConverters(TestBean3Converter::class)
    var testBean2: List<TestBean2>? = null
}